pytorch_lightning/callbacks/intel_compressor.py

```diff
@@ -0,0 +1,115 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Intel Quantization (TODO: pruning)
+^^^^^^^^^^^^
+
+"""
+from enum import Enum
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.core import LightningDataModule
+from pytorch_lightning.trainer.states import TrainerFn, TrainerStatus
+from pytorch_lightning.utilities import _NEURAL_COMPRESSOR_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+if _NEURAL_COMPRESSOR_AVAILABLE:
+    from neural_compressor.conf.config import Quantization_Conf
+    from neural_compressor.experimental import Quantization
+
+
+class QuantizationMode(Enum):

```

Comment:
@PenghuiCheng Why did you remove quantization aware training from here? Doesn't neural compressor have accuracy-driven tuning enhancement for it?

I have pinged you on Lightning Slack. I would like us to collaborate on this and land the PR by thinking of multiple approaches.

Reply:
@kaushikb11 , I'm glad to receive your suggestion, and let's collaborate on this PR. I removed the QAT approach because it seems INC processing is not matched PyTorch lightning's. So I removed QAT temporarily. And at the same time, we still continue to find ways to integrate methods for QAT. Your suggestions are very welcome. 
 BTW, How to use Lightning Slack? Could you give me a link?

Reply:
@PenghuiCheng Hey, use this [link](https://join.slack.com/t/pytorch-lightning/shared_invite/zt-pw5v393p-qRaDgEk24~EjiZNBpSQFgQ) to join Lightning Slack. You could ping me at Kaushik B.